hackathon_site/review/test_admin.py

```diff
@@ -0,0 +1,280 @@
+from django.contrib.auth.models import Permission
+from django.contrib.staticfiles.storage import staticfiles_storage
+from django.db.models import Q
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework import status
+
+from hackathon_site.tests import SetupUserMixin
+from registration.models import Team
+from review.models import Review
+
+static = staticfiles_storage.url
+
+
+class TeamReviewListAdminTestCase(SetupUserMixin, TestCase):
+    """
+    Test the list view for team reviews, which shows lists of teams and a few custom
+    fields and filters.
+    """
+
+    def setUp(self):
+        super().setUp()
+
+        self.list_view = reverse("admin:review_teamreview_changelist")
+        self.view_permissions = Permission.objects.filter(
+            Q(codename="view_application", content_type__app_label="registration")
+            | Q(codename="view_review", content_type__app_label="review"),
+        )
+        self.change_permissions = Permission.objects.filter(
+            Q(codename="view_application", content_type__app_label="registration")
+            | Q(codename="view_review", content_type__app_label="review")
+            | Q(codename="change_review", content_type__app_label="review"),
+        )
+
+        self.user.is_staff = True
+        self.user.save()
+
+    def _login(self, permissions=None):
+        super()._login()
+
+        if permissions is not None:
+            for perm in permissions:
+                self.user.user_permissions.add(perm)
+
+    def test_list_page_not_visible_without_permissions(self):
+        """
+        Without view permissions, the teams review page should not be visible
+        in the admin index list
+        """
+        self._login()
+        response = self.client.get(reverse("admin:index"))
+        self.assertNotContains(response, self.list_view)
+
+    def test_list_page_permission_denied(self):
+        """
+        Without view permissions, the teams review page should not be accessible
+        """
+        self._login()
+        response = self.client.get(self.list_view)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_list_page_shows_team_member_count(self):
```

Comment:
Can you add a small test with a team that's not full since a full team is kind of a degenerate case

Reply:
How so? .count() behaves the same whether there's 1 or 4 team members, so there's no reason to expect any difference since 4 isn't hard coded anywhere for the admin pages

Reply:
Changed this test to only have 3 team members to alleviate concerts about 4 being hard coded somewhere